challenge.rb

```diff
@@ -1,26 +1,30 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  result = []
+  input.each_index{ |idx| result.push(input[idx].capitalize)}
+  return result
 end
 
-def fetch_the_dog(input)
-  #implement your solution here
+def fetch_the_dog(input)	
+  return [input.detect{ |word| word == "dog"}]
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  return input.reject{ |word| word == "dog"}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  return input.length
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  first_two = []
+  (0..1).each{ |idx| first_two.push(input[idx])}
```

Comment:
See if [`Enumerable#first`](http://ruby-doc.org/core-2.2.2/Enumerable.html#method-i-first) leads to a simpler solution.